### PR TITLE
Binary parser for table and reference instructions

### PR DESCRIPTION
**Pull request by bbyalcinkaya**

### CODE REVIEW

File: wasm/opcodes/binary.py [hunk 1]
```diff
@@ -23,6 +23,17 @@ class BinaryOpcode(enum.Enum):
...
+    #
+    # Reference
+    #
+    REF_NULL = 0xD0
+    REF_IS_NULL = 0xD1
+    REF_FUNC = 0xD2
+
+    @property
+    def is_reference(self):
+        return 0xD0 <= self.value <= 0xD2
```
Comment (virgil-serbanuta) [inline wasm/opcodes/binary.py:35]: This is nice, I didn't know you could chain comparison operators.

File: wasm/parsers/instructions.py [hunk 2]
```diff
@@ -38,6 +41,13 @@ def parse_instruction(stream: IO[bytes]) -> Instruction:
     except IndexError:
         raise Exception("TODO: end of stream, what is the right exception here")
 
+    if opcode_value == 0xFC:
```
Comment (virgil-serbanuta) [inline wasm/parsers/instructions.py:46]: Could you add a comment explaining what 0xfc means?

File: wasm/parsers/tables.py [hunk 3]
```diff
@@ -49,3 +57,39 @@ def parse_table(stream: IO[bytes]) -> Table:
...
+    """
+
+    if opcode is BinaryOpcode.TABLE_GET:
+        tableidx = parse_table_idx(stream)
+        return TableGet(tableidx)
+    if opcode is BinaryOpcode.TABLE_SET:
+        tableidx = parse_table_idx(stream)
+        return TableSet(tableidx)
+    if opcode is BinaryOpcode.TABLE_INIT:
+        elemidx = parse_elem_idx(stream)
```
Comment (virgil-serbanuta) [inline wasm/parsers/tables.py:74]: Why is the elem first and the table second?

Comment (bbyalcinkaya) [reply wasm/parsers/tables.py:74]: That is how it is defined in the [spec](https://webassembly.github.io/spec/core/binary/instructions.html#table-instructions). I don't know why the parameters are flipped in the binary format. 

[Text format](https://webassembly.github.io/spec/core/text/instructions.html#table-instructions) (`=>` denotes t2a conversion):

```
plaininstr ::= 'table.init' x:tableIdx y:elemIdx => table.init x y
```

vs [Binary format](https://webassembly.github.io/spec/core/binary/instructions.html#table-instructions):

```
instr ::= 0xFC 12:u32 y:elemidx x:tableidx => table.init x y
```

**Review by virgil-serbanuta — APPROVED** [review]

**Pull request merged**